### PR TITLE
Add a Config Parser Interface

### LINKED ISSUE #92: Make a Parser interface
We did dabble with this when adam asked for specialized yml parsing, but I think this is a good idea to allow for more config file types.

### PR DESCRIPTION
**Pull request by SentryMan**
Now users can add custom parsers for handling different config file types. 

fixes #92

### CODE REVIEW

**Comment by SentryMan** [comment]
If you've got a better name for the interfaces feel free to change

File: avaje-config/src/main/java/io/avaje/config/InitialLoader.java [hunk 1]
```diff
@@ -283,57 +313,60 @@ CoreMap entryMap() {
...
-   */
-  private boolean loadYaml(String resourcePath, Source source) {
-    if (loadYamlPath(resourcePath + ".yaml", source)) {
-      return true;
-    } else {
-      return loadYamlPath(resourcePath + ".yml", source);
+  /** Load YAML first and if not found load YML. */
+ private boolean loadCustom(String resourcePath, Source source) {
+
+    for (var entry : parserMap.entrySet()) {
```
Comment (rbygrave) [inline avaje-config/src/main/java/io/avaje/config/InitialLoader.java:325]: This looks like it should just use the extension as map key

File: avaje-config/src/main/java/io/avaje/config/InitialLoader.java [hunk 2]
```diff
@@ -221,11 +243,9 @@ private void loadViaProfiles(Source source) {
     if (profiles != null) {
       for (final String path : profiles) {
         final var profile = loadContext.eval(path);
-        if (source != RESOURCE || !profileResourceLoaded.contains(profile)) {
-          if (load("application-" + profile, source)) {
-            profileResourceLoaded.add(profile);
-          }
-        }
+        if ((source != RESOURCE || !profileResourceLoaded.contains(profile)) && load("application-" + profile, source)) {
+        profileResourceLoaded.add(profile);
```
Comment (rbygrave) [inline avaje-config/src/main/java/io/avaje/config/InitialLoader.java:247]: style indentation

**Review by rbygrave — APPROVED** [review]
I'll merge and make some adjustments

**Pull request merged**